clients/client-java/src/main/java/com/datastrato/gravitino/client/api/SupportsCatalog.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.client.api;
+
+import com.datastrato.gravitino.Catalog;
+import com.datastrato.gravitino.CatalogChange;
+import com.datastrato.gravitino.NameIdentifier;
+import com.datastrato.gravitino.annotation.Evolving;
+import com.datastrato.gravitino.exceptions.CatalogAlreadyExistsException;
+import com.datastrato.gravitino.exceptions.NoSuchCatalogException;
+import com.datastrato.gravitino.exceptions.NoSuchMetalakeException;
+import java.util.Map;
+
+/**
+ * The client-side interface for supporting catalogs. It includes methods for listing, loading,
+ * creating, altering and dropping catalogs.
+ */
+@Evolving
+public interface SupportsCatalog {
```

Comment:
Maybe we can choose a different name, also I think the `SupportsCatalog` in the API module seems not so useful, we can remove that one.

Reply:
Also I was thinking that maybe we don't need this interface, just implementing the related methods in Client, that should also be worked, right?

Reply:
This interface is only in the "client" module, which limites its scope. Yes it is not that useful, I can drop it directly. Put it here just to make it clear that what are the new methods introduced.